lib/http/chainable.rb

```diff
@@ -213,9 +211,9 @@ def auth(value)
     def basic_auth(opts)
       user  = opts.fetch(:user)
       pass  = opts.fetch(:pass)
-      creds = "#{user}:#{pass}"
+      strict_base64_encoded_creds = ["#{user}:#{pass}"].pack("m0")
```

Comment:
We probably can introduce `HTTP::Utils` (don't like `Utils` modules in general, but it seems like fitting in this case)` with `encode_base64` method to avoid repeating `pack`. That should make code a bit better.

Reply:
Or how about `HTTP::Encoding`, where there are probably a few other things you could stick there to provide a nicer facade for various bits in the standard library

Reply:
Oh, yeah! `HTTP::Encoding` is a much better namespace.

Reply:
While trying to implement this suggestion I stumbled upon a conflict with the build-in `Encoding` constant. Each usage of the building encoding would need to be prefixed with `::` to make ruby look in the global namespace, in addition to required changes in other gems of the ecosystem like `http-form-data`.

I've went with `Encoder` instead which doesn't have such issues.

Reply:
Just saw this comment. Hm, I don't like `Encoder` name though. Probably we can go with `Base64` LOL:

``` ruby
module HTTP
  module Base64
    module_function

    def encode64(input)
      [input].pack("m0")
    end
  end

  class Request
    include Base64

    ...

      digest = encode64("#{proxy[:proxy_username]}:#{proxy[:proxy_password]}")

    ...
  end
end


```

Reply:
@tarcieri WDYT?

Reply:
This does get the tests all passing, however still suffer from the same (though now a bit more theoretical) problem. Other gems that extend the HTTP namespace would now get a different constant when acessing plain `Base64`.

Luckily the test suite exposed the problem with using `Encoding`. I can't say if that would be an issue with `Base64`.